tifeatures/layer.py

```diff
@@ -385,85 +461,50 @@ async def query(
         geom: str = None,
         dt: str = None,
         limit: Optional[int] = None,
+        geometry_column: Optional[GeometryColumn] = None,
         offset: Optional[int] = None,
         bbox_only: Optional[bool] = None,
         simplify: Optional[float] = None,
+        media_type: Optional[MediaType] = None,
     ) -> Tuple[FeatureCollection, int]:
         """Build and run Pg query."""
-        if geom and geom.lower() != "none" and not self.get_geometry_column(geom):
-            raise InvalidGeometryColumnName(f"Invalid Geometry Column: {geom}.")
-
-        sql_query = """
-            WITH
-                features AS (
-                    :features_q
-                ),
-                total_count AS (
-                    :count_q
-                )
-            SELECT json_build_object(
-                'type', 'FeatureCollection',
-                'features',
-                    (
-                        SELECT
-                            json_agg(
-                                json_build_object(
-                                    'type', 'Feature',
-                                    'id', :id_column,
-                                    'geometry', :geometry_q,
-                                    'properties', to_jsonb( features.* ) - :geom_columns::text[]
-                                )
-                            )
-                        FROM features
-                    ),
-                'total_count',
-                    (
-                        SELECT count FROM total_count
-                    )
-                )
-            ;
-        """
-        id_column = logic.V(self.id_column) or pg_funcs.cast(None, "text")
-        geom_columns = [g.name for g in self.geometry_columns]
-        q, p = render(
-            sql_query,
-            features_q=self._features_query(
+        count = await self._features_count_query(
+            pool=pool,
+            ids_filter=ids_filter,
+            datetime_filter=datetime_filter,
+            bbox_filter=bbox_filter,
+            properties_filter=properties_filter,
+            cql_filter=cql_filter,
+            geom=geom,
+            dt=dt,
+        )
```

Comment:
I think we should fully split the count and features class methods, because right now we do a count query for simple `feature` endpoint 🤷